### PR TITLE
New Code Example: Fine-tuning Wav2Vec 2.0 using CTC for ASR using Hugging Face Transformers

### PR DESCRIPTION
**Pull request by Sreyan88**
This code example shows how to fine-tune Wav2Vec 2.0 using CTC for ASR using Hugging Face Transformers. Additionally, it also shows best practices for Hugging Face Datasets.

This is part of my GSoC 2022 contribution. More information about my project can be found on [this link](https://docs.google.com/document/d/1fLDLwIhnwDUz3uUV8RyUZiOlmTN9Uzy5ZuvI8iDDFf8/edit#heading=h.bvepijofrmt9) and [this link](https://summerofcode.withgoogle.com/programs/2022/projects/tLgwQubw)

Thank You!

### CODE REVIEW

File: examples/audio/wav2vec2_ctc_asr_hf.py [hunk 1]
```diff
@@ -0,0 +1,685 @@
+"""
+Title: Finetuning Wav2Vec 2.0 with Hugging Face Transformers.
+Author: Sreyan Ghosh
+Date created: 2022/07/10
+Last modified: 2022/07/10
+Description: Finetuning Wav2Vec 2.0 with Hugging Face Transformers on CTC.
```
Comment (fchollet) [inline examples/audio/wav2vec2_ctc_asr_hf.py:6]: The meaning of "on CTC" here is unclear, please improve the description

File: examples/audio/wav2vec2_ctc_asr_hf.py [hunk 2]
```diff
@@ -0,0 +1,685 @@
...
+import json
+import random
+import logging
+import numpy as np
+
+# import tensorflow as tf
+from tensorflow import keras
+import IPython.display as ipd
+from dataclasses import dataclass, field
+from typing import Any, Dict, List, Optional, Union
```
Comment (fchollet) [inline examples/audio/wav2vec2_ctc_asr_hf.py:78]: Instead `import typing`

File: examples/audio/wav2vec2_ctc_asr_hf.py [hunk 3]
```diff
@@ -0,0 +1,685 @@
...
+
+import json
+import random
+import logging
+import numpy as np
+
+# import tensorflow as tf
+from tensorflow import keras
+import IPython.display as ipd
+from dataclasses import dataclass, field
```
Comment (fchollet) [inline examples/audio/wav2vec2_ctc_asr_hf.py:77]: instead `import dataclasses`

File: examples/audio/wav2vec2_ctc_asr_hf.py [hunk 4]
```diff
@@ -0,0 +1,685 @@
...
+a look at the "Alignment" section [here](https://distill.pub/2017/ctc/).
+"""
+
+vocab_dict["|"] = vocab_dict[" "]
+del vocab_dict[" "]
+
+vocab_dict["[UNK]"] = len(vocab_dict)
+vocab_dict["[PAD]"] = len(vocab_dict)
+
+vocab_dict
```
Comment (fchollet) [inline examples/audio/wav2vec2_ctc_asr_hf.py:293]: Use `print()`

File: examples/audio/wav2vec2_ctc_asr_hf.py [hunk 5]
```diff
@@ -0,0 +1,685 @@
...
+"""
+### Pre-process Audio Data
+
+So far, we have not looked at the actual values of the speech signal but just the
+transcription. In addition to `'text'`, our datasets include two more column names
+`'file'` and `'audio'`. `'file'` states the absolute path of the audio file. Let's take a
+look.
+"""
+
+librispeech["train"][0]["file"]
```
Comment (fchollet) [inline examples/audio/wav2vec2_ctc_asr_hf.py:332]: Likewise, etc

**Review by fchollet — COMMENTED** [review]
Thanks for the PR!

**Comment by pcoet** [comment]
Hi @Sreyan88, thanks for the PR. Are you planning to make changes in response to the feedback? Let us know if you're still working on this. Otherwise we'll close the request. Thanks!

**Comment by Sreyan88** [comment]
Hi @pcoet , My latest PR resolves all issues addressed by @fchollet . This can be merged. Thank You!

**Pull request closed without merging**